pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -409,8 +414,24 @@ async function resolveDependenciesOfImporters (
       const postponedPeersResolutionQueue: PostponedPeersResolutionFunction[] = []
       const pkgAddresses: PkgAddress[] = []
 
-      const resolvedDependenciesOfImporter = await Promise.all(
-        extendedWantedDeps.map((extendedWantedDep) => resolveDependenciesOfDependency(
+      async function resolveDependenciesOfImporter (extendedWantedDep: ExtendedWantedDependency) {
+        // The catalog protocol is only usable in importers (i.e. packages in
+        // the workspace. Replacing catalog protocol while resolving importers
+        // here before resolving dependencies of packages outside of the
+        // workspace/monorepo.
+        const catalogLookup = matchCatalogResolveResult(ctx.catalogResolver(extendedWantedDep.wantedDependency), {
+          found: (result) => result.resolution,
+          unused: () => undefined,
+          misconfiguration: (result) => {
+            throw result.error
+          },
+        })
+
+        if (catalogLookup != null) {
+          extendedWantedDep.wantedDependency.pref = catalogLookup.specifier
```

Comment:
I don't love the mutation happening here, but following a similar pattern in `resolveDependencies`, which we were discussing a few days ago.

https://github.com/pnpm/pnpm/blob/74c1057778c998a2af38a95d661be4f5f85e9e60/pkg-manager/resolve-dependencies/src/resolveDependencies.ts#L1145

Let me know if we should recreate the `extendedWantedDep` object instead.